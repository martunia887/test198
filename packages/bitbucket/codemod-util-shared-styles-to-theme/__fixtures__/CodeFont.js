// @flow
import { akCodeFontFamily as codeFont } from "@atlaskit/util-shared-styles";
import { N0 } from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-family: ${codeFont};
  vertical-align: top;
`;

//////
// @flow
import { codeFontFamily } from "@atlaskit/theme/constants";

import * as colors from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-family: ${codeFontFamily()};
  vertical-align: top;
`;
