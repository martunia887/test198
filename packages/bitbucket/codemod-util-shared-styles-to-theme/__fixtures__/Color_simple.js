// @flow
import IssueIcon from "@atlaskit/icon/glyph/issue";
import { akColorN30 } from "@atlaskit/util-shared-styles";

export const MyContainer = styled.div`
  color: ${akColorN30};
`;
//////
// @flow
import IssueIcon from "@atlaskit/icon/glyph/issue";
import { N30 } from "@atlaskit/theme/colors";

export const MyContainer = styled.div`
  color: ${N30};
`;
