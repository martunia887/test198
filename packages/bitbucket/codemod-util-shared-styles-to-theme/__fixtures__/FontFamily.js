// @flow
import { akFontFamily as font } from "@atlaskit/util-shared-styles";
import { N0 } from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-family: ${font};
  vertical-align: top;
`;

//////
// @flow
import { fontFamily } from "@atlaskit/theme/constants";

import * as colors from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-family: ${fontFamily()};
  vertical-align: top;
`;
