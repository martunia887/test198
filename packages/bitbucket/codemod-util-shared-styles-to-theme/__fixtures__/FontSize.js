// @flow
import { akFontSizeDefault as fontSize } from "@atlaskit/util-shared-styles";
import { N0 } from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-size: ${fontSize};
  vertical-align: top;
`;

//////
// @flow
import { fontSize } from "@atlaskit/theme/constants";

import * as colors from "@atlaskit/theme/colors";
import styled from "styled-components";

export const SourceLine = styled.td`
  background-color: ${N0};
  font-size: ${fontSize() + "px"};
  vertical-align: top;
`;