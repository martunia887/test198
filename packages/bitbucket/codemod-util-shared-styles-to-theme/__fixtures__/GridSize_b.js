// @flow
import { akGridSizeUnitless as gridSize } from "@atlaskit/util-shared-styles";
import styled from "styled-components";
import * as colors from "@atlaskit/theme/colors";

const blah = gridSize;
const MyDiv = styled.div`
  padding: ${gridSize}px;
`;
//////
// @flow
import styled from "styled-components";
import { gridSize } from "@atlaskit/theme/constants";
import * as colors from "@atlaskit/theme/colors";

const blah = gridSize();
const MyDiv = styled.div`
  padding: ${gridSize()}px;
`;
