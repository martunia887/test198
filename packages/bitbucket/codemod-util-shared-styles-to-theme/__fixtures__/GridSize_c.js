// @flow
import {
  akGridSizeUnitless,
  akTypographyMixins
} from "@atlaskit/util-shared-styles";
import styled from "styled-components";

export const EventContainer = styled.div`
  display: flex;
  margin-bottom: ${akGridSizeUnitless * 2}px;
`;

//////
// @flow
import { gridSize } from "@atlaskit/theme/constants";

import * as typography from "@atlaskit/theme/typography";
import styled from "styled-components";

export const EventContainer = styled.div`
  display: flex;
  margin-bottom: ${gridSize() * 2}px;
`;
