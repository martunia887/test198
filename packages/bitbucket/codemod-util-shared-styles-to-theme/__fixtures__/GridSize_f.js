// @flow
import { akGridSize, akGridSizeUnitless } from "@atlaskit/util-shared-styles";
import * as colors from "@atlaskit/theme/colors";
import styled, { css } from "styled-components";
const ComponentA = styled.div`
  padding: ${akGridSize};
  background: ${colors.N100};
  color: ${colors.P500};
`;
const ComponentB = styled.div`
  background: ${colors.N100};
  color: ${colors.P500};
  ${css`
    background: ${colors.N100};
    padding: ${akGridSizeUnitless}px;
    ${css`
      margin: ${akGridSize};
    `};
  `};
`;

//////
// @flow
import { gridSize } from "@atlaskit/theme/constants";

import * as colors from "@atlaskit/theme/colors";
import styled, { css } from "styled-components";
const ComponentA = styled.div`
  padding: ${gridSize()}px;
  background: ${colors.N100};
  color: ${colors.P500};
`;
const ComponentB = styled.div`
  background: ${colors.N100};
  color: ${colors.P500};
  ${css`
    background: ${colors.N100};
    padding: ${gridSize()}px;
    ${css`
      margin: ${gridSize()}px;
    `};
  `};
`;
