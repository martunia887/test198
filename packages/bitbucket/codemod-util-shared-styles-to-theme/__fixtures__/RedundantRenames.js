// @flow
import {
  akTypographyMixins as typo,
  akGridSizeUnitless as gridSize,
} from "@atlaskit/util-shared-styles";
import styled from "styled-components";
const gridSizePx = `${gridSize}px`;
const Fonty = styled.div`
  font-size: ${gridSizePx};
  ${typo.h100};
`;

//////
// @flow
import { gridSize } from "@atlaskit/theme/constants";

import { h100 } from "@atlaskit/theme/typography";
import styled from "styled-components";
const gridSizePx = `${gridSize()}px`;
const Fonty = styled.div`
  font-size: ${gridSizePx};
  ${h100()};
`;
