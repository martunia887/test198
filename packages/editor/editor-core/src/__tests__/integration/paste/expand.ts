import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { getDocFromElement, fullpage } from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
  copyAsHTML,
} from '../../__helpers/testing-example-helpers';
import { document } from './__fixtures__/document-with-table';

const editorSelector = '.ProseMirror';
const expandSelector = '[data-node-type="expand"]';
const nestedExpandSelector = '[data-node-type="nestedExpand"]';
const controlSelector = 'tbody tr:first-child th:nth-child(1)';

BrowserTestCase(
  'expand.ts: expand copied from renderer and pasted on full-page',
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const data =
      '<div id="RendererOutput"><div class="ak-renderer-document"><div data-node-type="expand" data-title="Expand title"><button aria-label="Expand Expand title"><p>Expand title</p></button><div><p>hello there</p></div></div></div></div></div>';
    await copyAsHTML(page, data);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      UNSAFE_allowExpand: true,
    });

    await page.click(fullpage.placeholder);
    await page.paste();
    await page.waitForSelector(expandSelector);

    const doc = await page.$eval(editorSelector, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'expand.ts: expand with legal content pasted in table',
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const data =
      '<div data-node-type="expand" data-title="title" data-pm-slice="0 0 []"><p><span data-mention-id="here" data-access-level="CONTAINER" contenteditable="false" data-user-type="SPECIAL">@here</span> hello</p></div>';
    await copyAsHTML(page, data);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(document),
      UNSAFE_allowExpand: true,
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);
    await page.paste();
    await page.waitForSelector(nestedExpandSelector);

    const doc = await page.$eval(editorSelector, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'expand.ts: expand with illegal content pasted in table',
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const data =
      '<div data-node-type="expand" data-title="title" data-pm-slice="0 0 []"><div data-panel-type="info"><div><p>content</p></div></div></div>';
    await copyAsHTML(page, data);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(document),
      UNSAFE_allowExpand: true,
      allowTables: {
        advanced: true,
      },
      allowPanel: true,
    });

    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);
    await page.paste();
    await page.waitForSelector(expandSelector);

    const doc = await page.$eval(editorSelector, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'expand.ts: nestedExpand pasted in table',
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const data =
      '<div data-node-type="nestedExpand" data-title="title" data-pm-slice="0 0 []"><p>hello there</p></div>';
    await copyAsHTML(page, data);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(document),
      UNSAFE_allowExpand: true,
      allowTables: {
        advanced: true,
      },
    });

    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);
    await page.paste();
    await page.waitForSelector(nestedExpandSelector);

    const doc = await page.$eval(editorSelector, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'expand.ts: nestedExpand pasted on top level',
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const data =
      '<div data-node-type="nestedExpand" data-title="title" data-pm-slice="0 0 []"><p>hello there</p></div>';
    await copyAsHTML(page, data);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      UNSAFE_allowExpand: true,
    });

    await page.click(fullpage.placeholder);
    await page.paste();
    await page.waitForSelector(expandSelector);

    const doc = await page.$eval(editorSelector, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
