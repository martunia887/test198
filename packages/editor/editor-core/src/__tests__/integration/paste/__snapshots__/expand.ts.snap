// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expand.ts: expand copied from renderer and pasted on full-page 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__expanded": false,
        "title": "Expand title",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "hello there",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "expand",
    },
  ],
  "type": "doc",
}
`;

exports[`expand.ts: expand with illegal content pasted in table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "attrs": Object {
        "__expanded": false,
        "title": "title",
      },
      "content": Array [
        Object {
          "attrs": Object {
            "panelType": "info",
          },
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "content",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "panel",
        },
      ],
      "type": "expand",
    },
  ],
  "type": "doc",
}
`;

exports[`expand.ts: expand with legal content pasted in table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__expanded": false,
                    "title": "title",
                  },
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "attrs": Object {
                            "accessLevel": "CONTAINER",
                            "id": "here",
                            "text": "@here",
                            "userType": "SPECIAL",
                          },
                          "type": "mention",
                        },
                        Object {
                          "text": " hello",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "nestedExpand",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`expand.ts: nestedExpand pasted in table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__expanded": false,
                    "title": "title",
                  },
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "hello there",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "nestedExpand",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`expand.ts: nestedExpand pasted on top level 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__expanded": false,
        "title": "title",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "hello there",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "expand",
    },
  ],
  "type": "doc",
}
`;
